### DISCUSSION TITLE
Bearer authentication in Uno?

### DISCUSSION THREAD

**Discussion by stevenroman760** [unoplatform/uno#8555] [Q&A] [answered]
Hello,

I have a server I need to call that is hosted in azure that has bearer authentication. When I try and request a token 

            app = ConfidentialClientApplicationBuilder.Create(config.ClientId)
                .WithClientSecret(config.ClientSecret)
                .WithAuthority(new Uri(config.Authority))
                .Build();

 result = app.AcquireTokenForClient(ResourceIds).ExecuteAsync().Result;

I get an exception on the first code block that says "{System.PlatformNotSupportedException: Confidential Client flows are not available on mobile platforms or on Mac.See https://aka.ms/msal-net-confidential-availability for details."

Do I need to have the server do a different kind of authorization scheme or is my code wrong?

Thanks,
Steven

**Comment by carldebilly** [comment, marked as answer]
I see two problems in your code:
1. You are using `.Result` on a `Task` instead of awaiting it. It can potentially lead to a deadlock and won't work at all on some platforms like WASM.  But that's not the cause of the problem you are describing.
2. You are using a `ConfidentialClientApplicationBuilder`, which is not supported on all platform by Microsoft [as described here](https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-client-applications). That's because some platforms are not considered secured enough to store secrets, so they must use the `PublicClientApplicationBuilder` instead. That's why you are getting the exception.
